crates/matrix-sdk/src/notification_settings/mod.rs

```diff
@@ -82,17 +82,11 @@ pub struct NotificationSettings {
     client: Client,
     /// Owner's account push rules. They will be updated on sync.
     rules: Arc<RwLock<Rules>>,
-    /// Event handler for push rules event
-    push_rules_event_handler: EventHandlerHandle,
+    /// Drop guard of event handler for push rules event.
+    _push_rules_event_handler_guard: Arc<EventHandlerDropGuard>,
```

Comment:
This is fine, but in my opinion it would also make sense to get rid of the `Clone` bound on this struct (and remove the `Arc` here).